Remove need for longgc and gcsimulator tests to have their own build

swgillespie opened the pull request:
Fixes #5083. This removes the dependency of long GC tests and GCSimulator tests on separate test builds.

swgillespie commented:
@dotnet-bot test CI please

swgillespie commented:
@dotnet-bot test CI please

swgillespie commented:
@dotnet-bot test Windows_NT Release longgc

swgillespie commented:
@ramarag PTAL?

Testing this more thoroughly today to make sure this works on both platforms, though it appears to be working fine now. The long GC PR failure here is due to the current version of netci.groovy invoking runtests.cmd with a command that got renamed in this PR, since PR jobs use the existing netci.groovy.

ramarag commented on netci.groovy:
```diff
@@ -1529,7 +1525,7 @@ combinedScenarios.each { scenario ->
                                         else if (architecture == 'x64') {
                                             if (isLongGc(scenario)) {
-                                                buildCommands += "tests\\runtest.cmd ${runtestArguments} longgctests sequential"
+                                                buildCommands += "tests\\runtest.cmd ${runtestArguments} ${scenario} sequential"
```
Just a suggestion can you move all the isLongGc checks before  https://github.com/dotnet/coreclr/pull/5621/files#diff-2d212ad84506951c8f928e21aed24580R1502

and then concatenate  ${scenario}  to ${runtestArguments}. Do you think something is preventing this ?

swgillespie replied:
I don't think so, no - I'll do that.

ramarag commented on tests/src/CLRTest.Execute.Bash.targets:
```diff
@@ -120,14 +120,39 @@ then
-echo "Skipping execution because long-running GC tests are not enabled"
-exit 0
+
+      <BashCLRTestGCLongTestSkipCondition Condition="'$(IsLongRunningGCTest)' != 'true'"><![CDATA[
```
It would also be nice to move this logic to another file like what was done for R2R in https://github.com/dotnet/coreclr/blob/master/tests/src/CLRTest.CrossGen.targets to say CLRTest.GC.targets

And append all the runtime logic to BashCLRTestPreCommands and CLRTestBatchPreCommands. You don't need a target to do this.  You can just use a property-group

swgillespie replied:
Ok!

ramarag commented:
@swgillespie Also list your scenario under https://github.com/dotnet/coreclr/blob/master/netci.groovy#L1910
so that redundant test builds are not triggered.

@wtgodbe : Is there a better way to group tests so that they do not trigger redundant builds, is it not getting too explicit or this is the way we have to live with ?

wtgodbe commented:
Maybe a constant list that maps from Scenario to InputScenario? Although it wouldn't be obvious to someone adding a new scenario that they should add their scenario to that map.

ramarag commented:
@wtgodbe : i was hoping to decide the logic somewhere in https://github.com/dotnet/coreclr/blob/master/netci.groovy#L1461 to redirect the job to a single job based on the arguments, instead of the job name already getting decided at this point. Does it make sense ? Is this even possible ?

I think this would be more intuitive

wtgodbe commented:
@ramarag I'm not sure I follow - the job name (which we use in flow jobs to pick up the proper build job) is based on the arguments (configuration, scenario, architecture, etc.), how are you suggesting we change this?

swgillespie commented:
@dotnet-bot test CI please

@ramarag - Addressed your feedback in the latest commit. For now I didn't try to generalize the input scenario - I'm happy to do it if you and Will have something in mind.

swgillespie commented on tests/src/CLRTest.GC.targets:
```diff
@@ -0,0 +1,105 @@
+    <BatchScriptSnippetGen>$(BatchScriptSnippetGen);GetGCBatchScript</BatchScriptSnippetGen>
+  </PropertyGroup>
+
+    <Target Name="GetGCBashScript">
```
It didn't, but I didn't spend very long troubleshooting it before moving to targets. I can spend some more time doing that if you'd like.

swgillespie commented:
@dotnet-bot test CI please

@ramarag updated, thanks!

ramarag commented:
LGTM, thanks for making the changes. Fixes https://github.com/dotnet/coreclr/issues/5083

swgillespie commented:
Squashed and merged. Thanks!
